plugins/inputs/http_response/http_response.go

```diff
@@ -130,15 +130,21 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 	// Start Timer
 	start := time.Now()
 	resp, err := h.client.Do(request)
+	fields["response_time"] = time.Since(start).Seconds()
+
 	if err != nil {
+		fields["http_response_code"] = -1
+		fields["response_string_match"] = 0
 		if h.FollowRedirects {
-			return nil, err
+			log.Printf("E! Failed to open HTTP connection: %s", err)
+			return fields, nil
```

Comment:
The reason I bring it up is 2 things. Doing it this way results in the log messages not having context. Meaning that if you have multiple inputs defined, you don't know which one is generating the error. The other problem is that I think we have an open issue somewhere to track statistics on per-input error counts, which requires errors going through `Accumulator.AddError()`.